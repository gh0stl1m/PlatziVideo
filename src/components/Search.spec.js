import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { cleanup, render, fireEvent, wait, waitForElement } from '../utils/test/test-utils';

// Component
import Search from './Search';

describe('<Search />', () => {
  afterEach(cleanup);

  it('Should render without crashing', () => {
    // Act
    const { container } = render(<Search />);

    // Asserts
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Given a list then the serch bar must be visible', () => {
    // Act
    const { getByText, getByPlaceholderText } = render(<Search />);

    // Asserts
    expect(getByText('¿Qué quieres ver hoy?')).toBeVisible();
    expect(getByPlaceholderText('Buscar...')).toBeVisible();
  });

  it('Given a list when an user click on the search and write down something, then the text must be appear', () => {
    // Arrange
    const inputSearchValue = "test";
    const props = {
     handlerChange: jest.fn(),
    }

    // Act
    const { getByPlaceholderText } = render(<Search {...props}/>);
    fireEvent.change(getByPlaceholderText('Buscar...'), { target: { value: inputSearchValue }});

    // Asserts
    expect(props.handlerChange).toHaveBeenCalled();
  });
});
