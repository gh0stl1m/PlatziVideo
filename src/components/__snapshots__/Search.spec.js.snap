// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Should render without crashing 1`] = `
<section
  class="main"
>
  <h2
    class="main__title"
  >
    ¿Qué quieres ver hoy?
  </h2>
  <input
    class="input"
    placeholder="Buscar..."
    type="text"
  />
</section>
`;
